feat(frontend): Implement tests for login hook

stevefaeembra opened the pull request:
Also needed a change for the render hook

stevefaeembra committed:
feat(frontend): Implement tests for login hook

Also needed a change for the render hook

marksmall commented on client/src/accounts/authentication/login.hook.test.tsx:
```diff
@@ -0,0 +1,68 @@
+import { act, renderHook, waitFor } from '~/test/utils';
+import { LoginForm } from './login-form.component';
+import { useLogin } from './login.hook';
+const ENDPOINT = '*/api/accounts/';
```
Don't you have a `./vscode/settings.json`, I expected eslint/prettier to complain about this, we have rules on the ordering of imports and to use whitespace to categorize imports. Also, the const right below it, there should definitely be whitespace to separate imports from other logical code blocks.

marksmall commented on client/src/accounts/authentication/login.hook.test.tsx:
```diff
@@ -0,0 +1,68 @@
+interface Result {
+  isSuccess: boolean;
+  isError: boolean;
+  data: Record<string, unknown>;
```
The shape of data returned should be known, we are verifying it in the hook, so I expect there is a type for it. Again, this is the point of types, if your not going to use them always, the code becomes a mess.

marksmall commented on client/src/accounts/authentication/login.hook.test.tsx:
```diff
@@ -0,0 +1,68 @@
+describe('useLoginHook', () => {
+  it('should throw an error for an unknown user', async () => {
+    const loginData = { email: 'bob@example.com', password: 'otherpassword', accessToken: 'foobar' };
+    server.use(rest.post(ENDPOINT, (req, res, ctx) => res(ctx.status(200), ctx.json({ message: 'User not found' }))));
```
I thought there was already some constants for HTTP codes in the frontend. If not, then there definitely should be. Lets get away from **magic numbers and strings**.

marksmall commented on client/src/accounts/authentication/login.hook.test.tsx:
```diff
@@ -0,0 +1,68 @@
+    act(() => {
+      result.current.mutate(loginData);
+    });
+    await waitFor(() => expect(result.current.isSuccess).toBe(false));
```
Try be consistent in your code, there is no whitespace here, but there is in other tests, such quirks are easy to introduce, we all do it, but this is why you have to create a checklist for yourself to help you catch the silly mistakes you personally make.

stevefaeembra committed:
fix(frontend): Fix vite config

Fixed vite/eslint config

stevefaeembra committed:
fix(frontend): Fix tests

Got rid of errors related to fetch.
Need to write tests in a very specific way to get them
to work
